server/src/test/java/org/elasticsearch/common/util/concurrent/ThreadContextTests.java

```diff
@@ -695,4 +696,172 @@ protected void doRun() throws Exception {
             }
         };
     }
+
+    public void testThreadNamingWithTracingIds() {
+        ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
+        Runnable withContext;
+
+        // an abstract runnable that has X-Opaque-Id in its header.
+        try (ThreadContext.StoredContext ignored = threadContext.stashContext()) {
+            threadContext.putHeader(Task.X_OPAQUE_ID, "some_id");
+            withContext = threadContext.preserveContext(new AbstractRunnable() {
+
+                @Override
+                public void onAfter() {
+                    assertFalse(Thread.currentThread().getName().contains("some_id"));
+                    assertFalse(Thread.currentThread().getName().contains(Task.X_OPAQUE_ID));
+                    assertFalse(Thread.currentThread().getName().contains(ThreadContext.WithTracingIdsInThreadName.ID_LABEL));
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    throw new RuntimeException("from onFailure", e);
+                }
+
+                @Override
+                protected void doRun() {
+                    // Assertions are errors, turn this into exception to make sure we get the error on failure
+                    try {
+                        assertTrue(Thread.currentThread().getName().contains("some_id"));
```

Comment:
Using `assertThat(name, containsString("some_id"));` would make any failures more readable and actionable, since we would see what failed to match

Reply:
Ah, good idea! Thanks!